cumulus/client/cirrus-executor/src/bundle_processor.rs

```diff
@@ -285,6 +285,57 @@ where
 			return Ok(None)
 		}
 
+		let best_execution_chain_number = self
+			.primary_chain_client
+			.runtime_api()
+			.best_execution_chain_number(&BlockId::Hash(primary_hash))?;
+
+		let best_execution_chain_number =
+			<NumberFor<Block>>::decode(&mut best_execution_chain_number.encode().as_slice())
+				.expect("Primary number and secondary number must use the same type; qed");
+
+		assert!(
+			header_number > best_execution_chain_number,
+			"Consensus chain number must larger than execution chain number by at least 1"
+		);
+
+		// Ideally, the receipt of current block will be included in the next block, i.e., no
+		// missing receipts.
+		if header_number == best_execution_chain_number + One::one() {
+			return self.try_sign_and_send_receipt(primary_hash, execution_receipt)
+		} else {
+			// Receipts for some previous blocks are missing.
+			let max_drift = self
+				.primary_chain_client
+				.runtime_api()
+				.maximum_receipt_drift(&BlockId::Hash(primary_hash))?;
+
+			let max_drift = <NumberFor<Block>>::decode(&mut max_drift.encode().as_slice())
+				.expect("Primary number and secondary number must use the same type; qed");
+
+			let max_allowed = (best_execution_chain_number + max_drift).min(header_number);
+
+			let mut to_send = best_execution_chain_number;
```

Comment:
Isn't `best_execution_chain_number` already present and doesn't need to be sent?

Reply:
Good catch!